src/Raven.Server/Documents/Indexes/Configuration/SingleIndexConfiguration.cs

```diff
@@ -29,7 +29,7 @@ public SingleIndexConfiguration(IndexConfiguration clientConfiguration, RavenCon
 
         public override bool RunInMemory => _databaseConfiguration.Indexing.RunInMemory;
 
-        public override PathSetting TempPath => _databaseConfiguration.Indexing.TempPath;
+        public override PathSetting TempPath => _databaseConfiguration.Indexing.TempPath ?? _databaseConfiguration.Indexing.StoragePath;
```

Comment:
I think it is better to append 'Temp' if you want to use Indexing.StoragePath to avoid putting files in the same dir?